File: 06_interesting_polygon/submissions/jimmy.js
let polygon_area = (n) => {
    return n**2 + (n-1)**2
}


let assert = require('assert')
assert.strictEqual(polygon_area(1), 1)
assert.strictEqual(polygon_area(2), 5)
assert.strictEqual(polygon_area(3), 13)
assert.strictEqual(polygon_area(4), 25)
console.log('All Tests Passed')


let full_test = () => {
    const start = new Date().getTime()
    let answers = require('./answers')
    answers.forEach(item => {
        let [n, answer] = item
        assert.strictEqual(polygon_area(n), answer)
    })
    console.log(`Full tests passed in: ${new Date().getTime() - start}ms`)
}

// Uncomment for full tests
// full_test()

File: 08_square_cypher/submissions/jimmy.py
import math
import re
import json
import time


def transposed(plaintext):
    plaintext = re.sub(' ', 'X', plaintext.upper())
    cypher_text = []
    char_list = re.findall(r'[A-Z]', plaintext)
    root = int(math.sqrt(len(char_list)))
    if len(char_list) > root ** 2:
        root += 1
    square = root ** 2
    while len(char_list) < square:
        char_list.append('X')
    index = 0
    while len(cypher_text) < len(char_list):
        cypher_text.append(char_list[index])
        index += root
        if index >= square:
            index -= square - 1
    return ''.join(cypher_text)


# Tests
assert transposed("The eagle flies at midnight!") == "TGIXGXHLEMHXEESITXXXXDXXEFANXXALTIXX"
assert transposed("The horse is in the barn, and it's a very big barn") == "TSNAXVGHEXRIEXEXTNTRBXIHXSYAHSEAXXROXXNABNRIBDXIX"
assert transposed("WECEXEXODXXDVXXAIEXXRSRXX") == "WEXAREXDISCOVEREDXXXXXXXX"
assert transposed("CROEMHOEXFITDETCDXENOONXXXXNIXGGDXGX") == "CODEXGREENXGOXTOXDEFCONXMIDNIGHTXXXX"
print("Initial tests passed")


def full_test():
    with open('answers.json') as f:
        answers = json.loads(f.read())
    start = time.time()
    for x, y in answers:
        assert transposed(x) == y
    print(f'All tests passed in {time.time() - start:.4f} seconds')


# Uncomment to test
full_test()
